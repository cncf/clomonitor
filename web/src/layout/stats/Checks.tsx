import { REPORT_OPTIONS } from '../../data';
import { ReportOption, ReportOptionData } from '../../types';
import sortChecks from '../../utils/sortChecks';
import ProgressBarInLine from './ProgressBarInLine';

interface Props {
  title: string;
  data: { [key in ReportOption]?: number };
  onSelectCheck: (name: ReportOption) => void;
}

function getOptionInfo(key: ReportOption) {
  return REPORT_OPTIONS[key];
}

const Checks = (props: Props) => {
  const sortedChecks = sortChecks(props.data);

  return (
    <div className="col-12">
      <div className="fw-bold border-bottom pb-2 mb-3 lightText">{props.title}</div>
      <div className="mx-1 pt-1">
        {sortedChecks.map((check: ReportOption) => {
          const opt: ReportOptionData = getOptionInfo(check);
          return (
            <ProgressBarInLine
              key={`check_${opt.name}`}
              title={opt.shortName || opt.name}
              name={check}
              icon={opt.icon}
              value={props.data[check] as number}
              onSelectCheck={props.onSelectCheck}
            />
          );
        })}
      </div>
    </div>
  );
};

export default Checks;
