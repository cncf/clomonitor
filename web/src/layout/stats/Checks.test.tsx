import { render, screen } from '@testing-library/react';

import Checks from './Checks';

const defaultProps = {
  data: {
    artifacthub_badge: 9,
    cla: 12,
    community_meeting: 43,
    dco: 83,
    ga4: 8,
    openssf_badge: 59,
    recent_release: 81,
    slack_presence: 33,
  },
  title: 'Best Practices',
  onSelectCheck: jest.fn(),
};

describe('Checks', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<Checks {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders component', () => {
      render(<Checks {...defaultProps} />);

      expect(screen.getByText('Best Practices')).toBeInTheDocument();
      expect(screen.getAllByText('Artifact Hub badge')).toHaveLength(2);
      expect(screen.getByText('9%')).toBeInTheDocument();
      expect(screen.getAllByText('CLA')).toHaveLength(2);
      expect(screen.getByText('12%')).toBeInTheDocument();
      expect(screen.getAllByText('Community meeting')).toHaveLength(2);
      expect(screen.getByText('43%')).toBeInTheDocument();
      expect(screen.getAllByText('DCO')).toHaveLength(2);
      expect(screen.getByText('83%')).toBeInTheDocument();
      expect(screen.getAllByText('Google Analytics 4')).toHaveLength(2);
      expect(screen.getByText('8%')).toBeInTheDocument();
      expect(screen.getAllByText('OpenSSF badge')).toHaveLength(2);
      expect(screen.getByText('59%')).toBeInTheDocument();
      expect(screen.getAllByText('Recent release')).toHaveLength(2);
      expect(screen.getByText('81%')).toBeInTheDocument();
      expect(screen.getAllByText('Slack presence')).toHaveLength(2);
      expect(screen.getByText('33%')).toBeInTheDocument();

      expect(screen.getAllByRole('progressbar')).toHaveLength(8);
    });

    it('renders checks properly sorted', () => {
      render(
        <Checks
          data={{ license_approved: 93, license_scanning: 22, license_spdx_id: 94 }}
          title="License"
          onSelectCheck={jest.fn()}
        />
      );

      expect(screen.getAllByText('License found')).toHaveLength(2);
      expect(screen.getAllByText('Approved license')).toHaveLength(2);
      expect(screen.getAllByText('License scanning')).toHaveLength(2);

      const progressbar = screen.getAllByRole('progressbar');
      expect(progressbar[0]).toHaveStyle({ width: '94%' });
      expect(progressbar[1]).toHaveStyle({ width: '93%' });
      expect(progressbar[2]).toHaveStyle({ width: '22%' });
    });
  });
});
