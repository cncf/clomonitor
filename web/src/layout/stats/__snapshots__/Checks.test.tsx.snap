// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-12"
  >
    <div
      class="fw-bold border-bottom pb-2 mb-3 lightText"
    >
      Best Practices
    </div>
    <div
      class="mx-1 pt-1"
    >
      <div
        class="d-flex flex-column wrapper"
      >
        <div
          class="d-flex flex-row align-items-center progressWrapper"
        >
          <div
            class="me-1 me-md-2 position-relative icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M332.8 320h38.4c6.4 0 12.8-6.4 12.8-12.8V172.8c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v134.4c0 6.4 6.4 12.8 12.8 12.8zm96 0h38.4c6.4 0 12.8-6.4 12.8-12.8V76.8c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v230.4c0 6.4 6.4 12.8 12.8 12.8zm-288 0h38.4c6.4 0 12.8-6.4 12.8-12.8v-70.4c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v70.4c0 6.4 6.4 12.8 12.8 12.8zm96 0h38.4c6.4 0 12.8-6.4 12.8-12.8V108.8c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v198.4c0 6.4 6.4 12.8 12.8 12.8zM496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
              />
            </svg>
          </div>
          <div
            class="progressTitle"
          >
            <span
              class="d-inline-block d-md-none"
            >
              Analytics
            </span>
            <span
              class="d-none d-md-inline-block"
            >
              <button
                aria-label="Search projects with passed Analytics check"
                class="btn btn-link p-0 btn"
              >
                Analytics
              </button>
            </span>
          </div>
          <div
            class="flex-grow-1 ms-2"
          >
            <div
              class="progress rounded-0 progress"
            >
              <div
                class="progress-bar progressbar"
                role="progressbar"
                style="width: 8%;"
              />
            </div>
          </div>
          <div
            class="ps-2 lh-1 text-end fw-bold lightText scoreWrapper"
          >
            8%
          </div>
        </div>
      </div>
      <div
        class="d-flex flex-column wrapper"
      >
        <div
          class="d-flex flex-row align-items-center progressWrapper"
        >
          <div
            class="me-1 me-md-2 position-relative icon"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 16V8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73l7 4a2 2 0 0 0 2 0l7-4A2 2 0 0 0 21 16z"
              />
            </svg>
          </div>
          <div
            class="progressTitle"
          >
            <span
              class="d-inline-block d-md-none"
            >
              Artifact Hub badge
            </span>
            <span
              class="d-none d-md-inline-block"
            >
              <button
                aria-label="Search projects with passed Artifact Hub badge check"
                class="btn btn-link p-0 btn"
              >
                Artifact Hub badge
              </button>
            </span>
          </div>
          <div
            class="flex-grow-1 ms-2"
          >
            <div
              class="progress rounded-0 progress"
            >
              <div
                class="progress-bar progressbar"
                role="progressbar"
                style="width: 9%;"
              />
            </div>
          </div>
          <div
            class="ps-2 lh-1 text-end fw-bold lightText scoreWrapper"
          >
            9%
          </div>
        </div>
      </div>
      <div
        class="d-flex flex-column wrapper"
      >
        <div
          class="d-flex flex-row align-items-center progressWrapper"
        >
          <div
            class="me-1 me-md-2 position-relative icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zM64 72c0-4.42 3.58-8 8-8h80c4.42 0 8 3.58 8 8v16c0 4.42-3.58 8-8 8H72c-4.42 0-8-3.58-8-8V72zm0 64c0-4.42 3.58-8 8-8h80c4.42 0 8 3.58 8 8v16c0 4.42-3.58 8-8 8H72c-4.42 0-8-3.58-8-8v-16zm192.81 248H304c8.84 0 16 7.16 16 16s-7.16 16-16 16h-47.19c-16.45 0-31.27-9.14-38.64-23.86-2.95-5.92-8.09-6.52-10.17-6.52s-7.22.59-10.02 6.19l-7.67 15.34a15.986 15.986 0 0 1-14.31 8.84c-.38 0-.75-.02-1.14-.05-6.45-.45-12-4.75-14.03-10.89L144 354.59l-10.61 31.88c-5.89 17.66-22.38 29.53-41 29.53H80c-8.84 0-16-7.16-16-16s7.16-16 16-16h12.39c4.83 0 9.11-3.08 10.64-7.66l18.19-54.64c3.3-9.81 12.44-16.41 22.78-16.41s19.48 6.59 22.77 16.41l13.88 41.64c19.77-16.19 54.05-9.7 66 14.16 2.02 4.06 5.96 6.5 10.16 6.5zM377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z"
              />
            </svg>
          </div>
          <div
            class="progressTitle"
          >
            <span
              class="d-inline-block d-md-none"
            >
              CLA
            </span>
            <span
              class="d-none d-md-inline-block"
            >
              <button
                aria-label="Search projects with passed CLA check"
                class="btn btn-link p-0 btn"
              >
                CLA
              </button>
            </span>
          </div>
          <div
            class="flex-grow-1 ms-2"
          >
            <div
              class="progress rounded-0 progress"
            >
              <div
                class="progress-bar progressbar"
                role="progressbar"
                style="width: 12%;"
              />
            </div>
          </div>
          <div
            class="ps-2 lh-1 text-end fw-bold lightText scoreWrapper"
          >
            12%
          </div>
        </div>
      </div>
      <div
        class="d-flex flex-column wrapper"
      >
        <div
          class="d-flex flex-row align-items-center progressWrapper"
        >
          <div
            class="me-1 me-md-2 position-relative icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M349.1 334.7c-11.2-4-29.5-4.2-37.6-7.3-5.6-2.2-14.5-4.6-17.4-8.1-2.9-3.5-2.9-28.5-2.9-28.5s7-6.6 9.9-14c2.9-7.3 4.8-27.5 4.8-27.5s6.6 2.8 9.2-10.4c2.2-11.4 6.4-17.4 5.3-25.8-1.2-8.4-5.8-6.4-5.8-6.4s5.8-8.5 5.8-37.4c0-29.8-22.5-59.1-64.6-59.1-42 0-64.7 29.4-64.7 59.1 0 28.9 5.7 37.4 5.7 37.4s-4.7-2-5.8 6.4c-1.2 8.4 3 14.4 5.3 25.8 2.6 13.3 9.2 10.4 9.2 10.4s1.9 20.1 4.8 27.5c2.9 7.4 9.9 14 9.9 14s0 25-2.9 28.5-11.8 5.9-17.4 8c-8 3.1-26.3 3.5-37.6 7.5-11.2 4-45.8 22.2-45.8 67.2h278.3c.1-45.1-34.5-63.3-45.7-67.3z"
              />
              <path
                d="M140 286s23.9-.8 33.4-9.3c-15.5-23.5-7.1-50.9-10.3-76.5-3.2-25.5-17.7-40.8-46.7-40.8h-.4c-28 0-43.1 15.2-46.3 40.8-3.2 25.5 5.7 56-10.2 76.5C69 285.3 93 285 93 285s1 14.4-1 16.8c-2 2.4-7.9 4.7-12 5.5-8.8 1.9-18.1 4.5-25.9 7.2-7.8 2.7-22.6 17.2-22.6 37.2h80.3c2.2-8 17.3-22.3 32-29.8 9-4.6 17.9-4.3 24.7-5.2 0 0 3.8-6-8.7-8.3 0 0-17.2-4.3-19.2-6.7-1.9-2.2-.6-15.7-.6-15.7zM372 286s-23.9-.8-33.4-9.3c15.5-23.5 7.1-50.9 10.3-76.5 3.2-25.5 17.7-40.8 46.7-40.8h.4c28 0 43.1 15.2 46.3 40.8 3.2 25.5-5.7 56 10.2 76.5-9.5 8.6-33.5 8.3-33.5 8.3s-1 14.4 1 16.8c2 2.4 7.9 4.7 12 5.5 8.8 1.9 18.1 4.5 25.9 7.2 7.8 2.7 22.6 17.2 22.6 37.2h-80.3c-2.2-8-17.3-22.3-32-29.8-9-4.6-17.9-4.3-24.7-5.2 0 0-3.8-6 8.7-8.3 0 0 17.2-4.3 19.2-6.7 1.9-2.2.6-15.7.6-15.7z"
              />
            </svg>
          </div>
          <div
            class="progressTitle"
          >
            <span
              class="d-inline-block d-md-none"
            >
              Community meeting
            </span>
            <span
              class="d-none d-md-inline-block"
            >
              <button
                aria-label="Search projects with passed Community meeting check"
                class="btn btn-link p-0 btn"
              >
                Community meeting
              </button>
            </span>
          </div>
          <div
            class="flex-grow-1 ms-2"
          >
            <div
              class="progress rounded-0 progress"
            >
              <div
                class="progress-bar progressbar"
                role="progressbar"
                style="width: 43%;"
              />
            </div>
          </div>
          <div
            class="ps-2 lh-1 text-end fw-bold lightText scoreWrapper"
          >
            43%
          </div>
        </div>
      </div>
      <div
        class="d-flex flex-column wrapper"
      >
        <div
          class="d-flex flex-row align-items-center progressWrapper"
        >
          <div
            class="me-1 me-md-2 position-relative icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M218.17 424.14c-2.95-5.92-8.09-6.52-10.17-6.52s-7.22.59-10.02 6.19l-7.67 15.34c-6.37 12.78-25.03 11.37-29.48-2.09L144 386.59l-10.61 31.88c-5.89 17.66-22.38 29.53-41 29.53H80c-8.84 0-16-7.16-16-16s7.16-16 16-16h12.39c4.83 0 9.11-3.08 10.64-7.66l18.19-54.64c3.3-9.81 12.44-16.41 22.78-16.41s19.48 6.59 22.77 16.41l13.88 41.64c19.75-16.19 54.06-9.7 66 14.16 1.89 3.78 5.49 5.95 9.36 6.26v-82.12l128-127.09V160H248c-13.2 0-24-10.8-24-24V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24v-40l-128-.11c-16.12-.31-30.58-9.28-37.83-23.75zM384 121.9c0-6.3-2.5-12.4-7-16.9L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1zm-96 225.06V416h68.99l161.68-162.78-67.88-67.88L288 346.96zm280.54-179.63l-31.87-31.87c-9.94-9.94-26.07-9.94-36.01 0l-27.25 27.25 67.88 67.88 27.25-27.25c9.95-9.94 9.95-26.07 0-36.01z"
              />
            </svg>
          </div>
          <div
            class="progressTitle"
          >
            <span
              class="d-inline-block d-md-none"
            >
              DCO
            </span>
            <span
              class="d-none d-md-inline-block"
            >
              <button
                aria-label="Search projects with passed DCO check"
                class="btn btn-link p-0 btn"
              >
                DCO
              </button>
            </span>
          </div>
          <div
            class="flex-grow-1 ms-2"
          >
            <div
              class="progress rounded-0 progress"
            >
              <div
                class="progress-bar progressbar"
                role="progressbar"
                style="width: 83%;"
              />
            </div>
          </div>
          <div
            class="ps-2 lh-1 text-end fw-bold lightText scoreWrapper"
          >
            83%
          </div>
        </div>
      </div>
      <div
        class="d-flex flex-column wrapper"
      >
        <div
          class="d-flex flex-row align-items-center progressWrapper"
        >
          <div
            class="me-1 me-md-2 position-relative icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 22c3.859 0 7-3.141 7-7s-3.141-7-7-7c-3.86 0-7 3.141-7 7s3.14 7 7 7zm0-12c2.757 0 5 2.243 5 5s-2.243 5-5 5-5-2.243-5-5 2.243-5 5-5zm-1-8H7v5.518a8.957 8.957 0 0 1 4-1.459V2zm6 0h-4v4.059a8.957 8.957 0 0 1 4 1.459V2z"
              />
              <path
                d="m10.019 15.811-.468 2.726L12 17.25l2.449 1.287-.468-2.726 1.982-1.932-2.738-.398L12 11l-1.225 2.481-2.738.398z"
              />
            </svg>
          </div>
          <div
            class="progressTitle"
          >
            <span
              class="d-inline-block d-md-none"
            >
              OpenSSF badge
            </span>
            <span
              class="d-none d-md-inline-block"
            >
              <button
                aria-label="Search projects with passed OpenSSF badge check"
                class="btn btn-link p-0 btn"
              >
                OpenSSF badge
              </button>
            </span>
          </div>
          <div
            class="flex-grow-1 ms-2"
          >
            <div
              class="progress rounded-0 progress"
            >
              <div
                class="progress-bar progressbar"
                role="progressbar"
                style="width: 59%;"
              />
            </div>
          </div>
          <div
            class="ps-2 lh-1 text-end fw-bold lightText scoreWrapper"
          >
            59%
          </div>
        </div>
      </div>
      <div
        class="d-flex flex-column wrapper"
      >
        <div
          class="d-flex flex-row align-items-center progressWrapper"
        >
          <div
            class="me-1 me-md-2 position-relative icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 0H2a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2zM1 3.857C1 3.384 1.448 3 2 3h12c.552 0 1 .384 1 .857v10.286c0 .473-.448.857-1 .857H2c-.552 0-1-.384-1-.857V3.857z"
              />
              <path
                d="M6.5 7a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm-9 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm-9 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2z"
              />
            </svg>
          </div>
          <div
            class="progressTitle"
          >
            <span
              class="d-inline-block d-md-none"
            >
              Recent release
            </span>
            <span
              class="d-none d-md-inline-block"
            >
              <button
                aria-label="Search projects with passed Recent release check"
                class="btn btn-link p-0 btn"
              >
                Recent release
              </button>
            </span>
          </div>
          <div
            class="flex-grow-1 ms-2"
          >
            <div
              class="progress rounded-0 progress"
            >
              <div
                class="progress-bar progressbar"
                role="progressbar"
                style="width: 81%;"
              />
            </div>
          </div>
          <div
            class="ps-2 lh-1 text-end fw-bold lightText scoreWrapper"
          >
            81%
          </div>
        </div>
      </div>
      <div
        class="d-flex flex-column wrapper"
      >
        <div
          class="d-flex flex-row align-items-center progressWrapper"
        >
          <div
            class="me-1 me-md-2 position-relative icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M94.12 315.1c0 25.9-21.16 47.06-47.06 47.06S0 341 0 315.1c0-25.9 21.16-47.06 47.06-47.06h47.06v47.06zm23.72 0c0-25.9 21.16-47.06 47.06-47.06s47.06 21.16 47.06 47.06v117.84c0 25.9-21.16 47.06-47.06 47.06s-47.06-21.16-47.06-47.06V315.1zm47.06-188.98c-25.9 0-47.06-21.16-47.06-47.06S139 32 164.9 32s47.06 21.16 47.06 47.06v47.06H164.9zm0 23.72c25.9 0 47.06 21.16 47.06 47.06s-21.16 47.06-47.06 47.06H47.06C21.16 243.96 0 222.8 0 196.9s21.16-47.06 47.06-47.06H164.9zm188.98 47.06c0-25.9 21.16-47.06 47.06-47.06 25.9 0 47.06 21.16 47.06 47.06s-21.16 47.06-47.06 47.06h-47.06V196.9zm-23.72 0c0 25.9-21.16 47.06-47.06 47.06-25.9 0-47.06-21.16-47.06-47.06V79.06c0-25.9 21.16-47.06 47.06-47.06 25.9 0 47.06 21.16 47.06 47.06V196.9zM283.1 385.88c25.9 0 47.06 21.16 47.06 47.06 0 25.9-21.16 47.06-47.06 47.06-25.9 0-47.06-21.16-47.06-47.06v-47.06h47.06zm0-23.72c-25.9 0-47.06-21.16-47.06-47.06 0-25.9 21.16-47.06 47.06-47.06h117.84c25.9 0 47.06 21.16 47.06 47.06 0 25.9-21.16 47.06-47.06 47.06H283.1z"
              />
            </svg>
          </div>
          <div
            class="progressTitle"
          >
            <span
              class="d-inline-block d-md-none"
            >
              Slack presence
            </span>
            <span
              class="d-none d-md-inline-block"
            >
              <button
                aria-label="Search projects with passed Slack presence check"
                class="btn btn-link p-0 btn"
              >
                Slack presence
              </button>
            </span>
          </div>
          <div
            class="flex-grow-1 ms-2"
          >
            <div
              class="progress rounded-0 progress"
            >
              <div
                class="progress-bar progressbar"
                role="progressbar"
                style="width: 33%;"
              />
            </div>
          </div>
          <div
            class="ps-2 lh-1 text-end fw-bold lightText scoreWrapper"
          >
            33%
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
